src/components/BulkEnrollmentPage/stepper/AddCoursesStep.tsx

```diff
@@ -0,0 +1,171 @@
+  enterpriseId,
+  enterpriseSlug,
+  selectedCourses,
+  subscription,
+  algolia,
+}) => {
+  const algoliaFilters = useAlgoliaFilters(subscription, algolia);
+  return (
+    <>
+      <DismissibleCourseWarning defaultShow={(selectedCourses?.length || 0) > MAX_COURSES} />
+      <SearchData>
+        <InstantSearch
+          indexName={configuration.ALGOLIA.INDEX_NAME}
+          searchClient={algolia.searchClient}
+        >
+          <Configure
+            filters={algoliaFilters}
+            hitsPerPage={NUM_CONTENT_ITEMS_PER_PAGE}
+          />
+          <SearchHeader variant="default" />
+          <CourseSearchResults
+            enterpriseId={enterpriseId}
+            enterpriseSlug={enterpriseSlug}
+            subscriptionUUID={subscription.uuid}
+          />
+        </InstantSearch>
+      </SearchData>
+    </>
+  );
+};
+
+const mapStateToProps = (state) => ({
+  enterpriseId: state.portalConfiguration.enterpriseId,
+  enterpriseSlug: state.portalConfiguration.enterpriseSlug,
+});
+
+const SearchEnabled = connect(mapStateToProps)(BaseSearchEnabled);
+
+interface AddCoursesStepProps {
+  subscription: Subscription;
+  algolia: UseAlgoliaSearchResult;
+}
+
+export const BaseAddCoursesStep: React.FC<AddCoursesStepProps> = ({
+  subscription,
+  algolia,
+}) => {
+  const {
+    courses: [selectedCourses],
+  } = useContext(BulkEnrollContext);
+
+  const catalogQueryUuid = algolia.catalogUuidsToCatalogQueryUuids?.[subscription.enterpriseCatalogUuid];
+
+  useEffect(() => {
+    if (!algolia.isCatalogQueryFiltersEnabled || algolia.isLoading) {
+      // Filtering Algolia by catalog query is not enabled or the
+      // query is still loading; skipping.
+      return;
+    }
+
+    if (!catalogQueryUuid) {
+      // Log an error if the catalog query UUID is not found.
+      const error = new Error(`No catalog query UUID found for the subscription ${subscription.uuid} and its catalog ${subscription.enterpriseCatalogUuid}`);
+      logError(error);
+    }
+  }, [
+    catalogQueryUuid,
+    subscription.uuid,
+    subscription.enterpriseCatalogUuid,
+    algolia.isCatalogQueryFiltersEnabled,
+    algolia.isLoading,
+  ]);
+
+  if (algolia.isCatalogQueryFiltersEnabled) {
+    if (algolia.isLoading) {
+      return (
+        <BaseAddCoursesStepContents>
+          <Skeleton height={360} />
+          <div data-testid="skeleton-algolia-loading-courses" className="sr-only">Loading courses...</div>
+        </BaseAddCoursesStepContents>
+      );
+    }
+    return (
+      <BaseAddCoursesStepContents>
+        <SearchEnabled
+          selectedCourses={selectedCourses}
+          algolia={algolia}
+          subscription={subscription}
+        />
+      </BaseAddCoursesStepContents>
+    );
+  }
+
+  return (
+    <BaseAddCoursesStepContents>
+      <SearchEnabled
+        selectedCourses={selectedCourses}
+        algolia={algolia}
+        subscription={subscription}
+      />
+    </BaseAddCoursesStepContents>
+  );
+};
```

Comment:
[nit] This could be simplified a bit

``` jsx
  if (algolia.isCatalogQueryFiltersEnabled && algolia.isLoading) {
      return (
        <BaseAddCoursesStepContents>
          <Skeleton height={360} />
          <div data-testid="skeleton-algolia-loading-courses" className="sr-only">Loading courses...</div>
        </BaseAddCoursesStepContents>
      );
  }

  return (
    <BaseAddCoursesStepContents>
      <SearchEnabled
        selectedCourses={selectedCourses}
        algolia={algolia}
        subscription={subscription}
      />
    </BaseAddCoursesStepContents>
  );
```

Reply:
good catch! will update